### PR TITLE
Feature data queries

### PR DESCRIPTION
**Pull request by ishimwepati**
Hi!

Below are what I implemented :

1. I added a column species of type string to my animals table,
2. I inserted new data,
3. Inside a transaction I updated the animals table by setting the species column to unspecified:
Screenshot 2023-09-29 at 19 23 00
Screenshot 2023-09-29 at 19 23 43

4. Inside a transaction I did : 
Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Verify that changes were made.
Commit the transaction.
Verify that changes persist after commit.

Screenshot 2023-09-29 at 19 26 13
Screenshot 2023-09-29 at 19 26 22
Screenshot 2023-09-29 at 19 27 14

5. Inside a transaction I delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, I can start breathing as usual ;)
Screenshot 2023-09-29 at 19 29 39

6. Below are what I did next :
Screenshot 2023-09-29 at 19 37 01

7. Differents querries 👍 

-- Count the total number of animals in the table
Screenshot 2023-09-29 at 19 39 17

-- Count the animals with zero escape attempts
Screenshot 2023-09-29 at 19 40 55

-- Calculate the average weight of all animals
Screenshot 2023-09-29 at 19 41 54

-- Identify whether neutered or not neutered animals have the most escape attempts
Screenshot 2023-09-29 at 19 43 30

-- Find the minimum and maximum weight for each type of animal (by species)
Screenshot 2023-09-29 at 19 44 35

-- Calculate the average escape attempts for animals born between 1990 and 2000, grouped by species
Screenshot 2023-09-29 at 19 45 37

 I added screenshots with the results of my queries and with the results of my play with transactions to this Pull Request description.
 
 Your suggestions are most welcome !

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -8,9 +8,10 @@ CREATE TABLE animals (
...
-    weight_kg decimal );
+    weight_kg decimal, 
+    species text );
 
 TABLESPACE pg_default;
 
 ALTER TABLE IF EXISTS public.animals
-    OWNER to wazacode;
\ No newline at end of file
+    OWNER to wazacode;
```
Comment (TadesseDev) [inline schema.sql]: - [x] Dear @ishimwepati, here we are trying to alter the `animal` table and add a column named species. I can see you alter the owner, but this is not what is required in this milestone. Please apply `alter` and add the `species` column.  
**Note: I see you have added the `species` column as part of the create table, But we need to use the `alter table` to add this column**
> This is to help you learn how to add a column after the table is created.

File: queries.sql [hunk 2]
```diff
@@ -23,4 +22,83 @@ SELECT * from animals WHERE neutered = true;
...
+
+-- Start a transaction
+BEGIN; 
+
+DELETE FROM animals;
+
+SELECT * FROM animals;
+
+ROLLBACK; 
+
```
Comment (TadesseDev) [inline queries.sql]: - [Optional] Please add a select statement after the `ROLLBACK;` here, this will help us check if the transaction is indeed canceled.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @ishimwepati 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Most of the requirement has been meet. ✅
 - [x]  Data has been inserted appropriately ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (ishimwepati) [reply schema.sql]: Thanks a lot  for this !

File: queries.sql (hunk 2 above)
Comment (ishimwepati) [reply queries.sql]: Thanks for this suggestion !
I have worked on it !

**Review by ademibowale — APPROVED** [review]
Hi @ishimwepati  , 


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️
-  Add a new column called species with a string type ✔️
- Everything that creates or change the tables structure should be saved in the schema.sql file..✔️
- Everything that is a query that helps to analyze data should be saved in the queries.sql file.✔️
- Descriptive commit messages ✔️ 
- Good use of Github flow ✔️






- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @Adebowale ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ishimwepati** [comment]
Thanks !

**Pull request merged**